kotlin/workflow-runtime/src/main/java/com/squareup/workflow/internal/WorkflowNode.kt

```diff
@@ -119,7 +120,7 @@ internal class WorkflowNode<InputT, StateT, OutputT : Any, RenderingT>(
     handler: (OutputT) -> T?
   ) {
     fun acceptUpdate(action: WorkflowAction<StateT, OutputT>): T? {
-      val (newState, output) = action(state)
+      val (newState, output) = action.fire(state)
```

Comment:
Could save a Pair allocation by creating the Mutator here directly to close over the state property. This prevents any leaks of the sink from surprisingly mutating the property though so the safety is nice.

Reply:
I'm inclined to err on the side of safety. That's why I like the `Mutator` API in general. We can always optimize if this somehow turns out to be a hot spot.